x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -90,11 +92,16 @@ class RollupShardIndexer {
     private final String[] metricFields;
     private final String[] labelFields;
     private final Map<String, FieldValueFetcher> fieldValueFetchers;
-    private final AtomicLong numSent = new AtomicLong();
-    private final AtomicLong numIndexed = new AtomicLong();
-    private final AtomicLong numFailed = new AtomicLong();
+    private final AtomicLong numReceived;
```

Comment:
Maybe use the use getters on the `task` object directly? Instead of also defining these fields here.

Reply:
There is a reference to the `task` here. So I think the atomic long fields can be removed here and
instead the atomic longs from the `RollupShardTask` can be used directly in this class.

Reply:
Can you also make this (^) change?